core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -564,6 +567,10 @@ object Doc {
    */
   private[paiges] case class Align(doc: Doc) extends Doc
 
+  private[paiges] case class LazyDoc(generate: () => Doc) extends Doc {
```

Comment:
Why `() => Doc` as opposed to `Unit => Doc`?  I've never understood the former and chalked it up to "some weird Scala thing" compared with the later, which actually looks like a type.

Reply:
Ah thanks!  I tend to forget that `=>` is syntactic sugar.  The `Function0` comment makes it very clear.